### PR TITLE
Sketcher_Dimension: Adds arc angle mode.

### PR DESCRIPTION
**Pull request by PaddleStroke**
This PR adds arc angle mode to sketcher_dimension.
It also adds a tiny convenience function, cf commits.

Fixes https://github.com/FreeCAD/FreeCAD/issues/10605

### CODE REVIEW

File: src/Mod/Part/App/Geometry.cpp [hunk 1]
```diff
@@ -2422,6 +2422,19 @@ void GeomArcOfCircle::setRadius(double Radius)
     }
 }
 
+/*!
+ * \brief GeomArcOfCircle::getAngle
+ * \param emulateCCWXY: if true, the arc will pretent to be a CCW arc in XY plane.
```
Comment (luzpaz) [inline src/Mod/Part/App/Geometry.cpp:2427]: `s/pretent/pretend`

**Comment by PaddleStroke** [comment]
Typo fixed

**Comment by howie-j** [comment]
Did a quick test. Works great and solves the issue. Thanks :+1: 

angle

**Comment by chennes** [comment]
@PaddleStroke this is probably ready for a merge once the conflict is resolved.

**Comment by PaddleStroke** [comment]
@chennes thanks. I rebased. Cool to see isArcOfCircle() and others helper merged.

**Comment by PaddleStroke** [comment]
Force pushed to replace typeId test by isArcOfCircle

**Pull request merged**